network/hostnetwork/dht.go

```diff
@@ -60,7 +60,7 @@ type DHT struct {
 	subnet            Subnet
 	timeout           int // bootstrap reconnect timeout
 	infinityBootstrap bool
-	nodeID            core.RecordRef
+	nodeID            *core.RecordRef
```

Comment:
I don't like this change; i think we should be consistent across code base and use everywhere RecordRef references by value. Passing RecordRef by pointer should be used if we want to handle cases when we pass nil instead (not this case)